pallets/nfts/src/weights.rs

```diff
@@ -549,43 +548,131 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
+		// Proof Size summary in bytes:
+		//  Measured:  `450`
+		//  Estimated: `3602`
+		// Minimum execution time: 20_000_000 picoseconds.
+		Weight::from_parts(21_000_000, 3602)
+			.saturating_add(T::DbWeight::get().reads(3_u64))
+			.saturating_add(T::DbWeight::get().writes(1_u64))
+	}
 	/// Storage: `Nfts::Item` (r:1 w:1)
 	/// Proof: `Nfts::Item` (`max_values`: None, `max_size`: Some(861), added: 3336, mode: `MaxEncodedLen`)
+	/// Storage: `Nfts::CollectionApprovals` (r:1 w:0)
+	/// Proof: `Nfts::CollectionApprovals` (`max_values`: None, `max_size`: Some(137), added: 2612, mode: `MaxEncodedLen`)
 	fn cancel_approval() -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `418`
+		//  Measured:  `345`
 		//  Estimated: `4326`
-		// Minimum execution time: 16_080_000 picoseconds.
-		Weight::from_parts(16_603_000, 4326)
+		// Minimum execution time: 11_000_000 picoseconds.
+		Weight::from_parts(12_000_000, 4326)
+			.saturating_add(T::DbWeight::get().reads(2_u64))
+			.saturating_add(T::DbWeight::get().writes(1_u64))
+	}
+	/// Storage: `Nfts::CollectionApprovals` (r:1 w:1)
+	/// Proof: `Nfts::CollectionApprovals` (`max_values`: None, `max_size`: Some(137), added: 2612, mode: `MaxEncodedLen`)
+	fn cancel_collection_approval() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `359`
+		//  Estimated: `3602`
+		// Minimum execution time: 16_000_000 picoseconds.
+		Weight::from_parts(17_000_000, 3602)
+			.saturating_add(T::DbWeight::get().reads(1_u64))
+			.saturating_add(T::DbWeight::get().writes(1_u64))
+	}
+	/// Storage: `Nfts::CollectionApprovals` (r:1 w:1)
+	/// Proof: `Nfts::CollectionApprovals` (`max_values`: None, `max_size`: Some(137), added: 2612, mode: `MaxEncodedLen`)
+	fn force_cancel_collection_approval() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `359`
+		//  Estimated: `3602`
+		// Minimum execution time: 17_000_000 picoseconds.
+		Weight::from_parts(17_000_000, 3602)
 			.saturating_add(T::DbWeight::get().reads(1_u64))
 			.saturating_add(T::DbWeight::get().writes(1_u64))
 	}
 	/// Storage: `Nfts::Item` (r:1 w:1)
 	/// Proof: `Nfts::Item` (`max_values`: None, `max_size`: Some(861), added: 3336, mode: `MaxEncodedLen`)
+	/// Storage: `Nfts::CollectionApprovals` (r:1 w:0)
+	/// Proof: `Nfts::CollectionApprovals` (`max_values`: None, `max_size`: Some(137), added: 2612, mode: `MaxEncodedLen`)
 	fn clear_all_transfer_approvals() -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `418`
+		//  Measured:  `345`
 		//  Estimated: `4326`
-		// Minimum execution time: 15_013_000 picoseconds.
-		Weight::from_parts(15_607_000, 4326)
-			.saturating_add(T::DbWeight::get().reads(1_u64))
+		// Minimum execution time: 12_000_000 picoseconds.
+		Weight::from_parts(12_000_000, 4326)
+			.saturating_add(T::DbWeight::get().reads(2_u64))
 			.saturating_add(T::DbWeight::get().writes(1_u64))
 	}
+	/// Storage: `Nfts::CollectionApprovals` (r:1000 w:999)
```

Comment:
Curious why this is not:
```suggestion
	/// Storage: `Nfts::CollectionApprovals` (r:1000 w:1000)
```

Reply:
Are we sure that `drain_prefix` not first reads all values and then it is iterated over the values. In stead of what we want, reading the storage map per value and thus stop when we reach the limit?

Reply:
Could be because `n` iterates from `1..1_000` and `approve_collection_transfer` is called from `0..n`.

> Are we sure that drain_prefix not first reads all values and then it iterates over the values?

`drain_prefix` is also used for a similar purpose in the `do_cancel_item_attributes_approval` so this could not be a case. I'm validating...

Reply:
Alright thank you